nautilus_trader/common/actor.pyx

```diff
@@ -2079,7 +2079,7 @@ cdef class Actor(Component):
         datetime end = None,
         int limit = 0,
         callback: Callable[[UUID4], None] | None = None,
-        bint update_catalog = False,
+        str update_catalog_mode = None,
```

Comment:
We could also create a `CatalogWriteMode` enum (or similar)? just to lock the typing and chance of errors down a little more. A regular Python enum would work (doesn't have to be one which starts in Rust and needs 1000 lines of code).